benchmarks/union_find/union_find.cpp

```diff
@@ -0,0 +1,167 @@
+  {
+    // Construct random permutation by sorting a vector of random values
+    Kokkos::View<int *, MemorySpace> random_values(
+        Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                           "ArborX::Benchmark::random_values"),
+        num_edges);
+    Kokkos::parallel_for(
+        "ArborX::Bechmark::init_random_values",
+        Kokkos::RangePolicy<ExecutionSpace>(ExecutionSpace{}, 0, num_edges),
+        KOKKOS_LAMBDA(int i) {
+          auto rand_gen = rand_pool.get_state();
+          random_values(i) = rand_gen.rand();
+          rand_pool.free_state(rand_gen);
+        });
+    auto permute = ArborX::Details::sortObjects(exec_space, random_values);
+
+    // Init edges in a random order
+    Kokkos::parallel_for(
+        "ArborX::Bechmark::init_edges",
+        Kokkos::RangePolicy<ExecutionSpace>(ExecutionSpace{}, 0, num_edges),
+        KOKKOS_LAMBDA(unsigned i) {
+          auto rand_gen = rand_pool.get_state();
+          edges(permute(i)) = {rand_gen.urand() % (i + 1), i + 1};
+          rand_pool.free_state(rand_gen);
+        });
+  }
+  return edges;
+}
+
+template <typename ExecutionSpace>
+auto buildUnionFind(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::View<int *, MemorySpace> labels(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::labels"),
+      n);
+  ArborX::iota(exec_space, labels);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if constexpr (std::is_same_v<ExecutionSpace, Kokkos::Serial>)
+    return ArborX::Details::UnionFind<MemorySpace, true>(labels);
+  else
+#endif
+    return ArborX::Details::UnionFind<MemorySpace, false>(labels);
+}
+
+template <typename ExecutionSpace, typename Edges, typename UnionFind>
+double benchmark(ExecutionSpace const &exec_space, Edges edges,
+                 UnionFind union_find)
+{
+  Kokkos::fence();
+  Kokkos::Timer timer;
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::union-find",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, edges.size()),
+      KOKKOS_LAMBDA(int e) {
+        int i = edges(e).source;
+        int j = edges(e).target;
+
+        union_find.merge(i, j);
+      });
+  Kokkos::fence();
+  return timer.seconds();
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+
+  std::cout << "ArborX version    : " << ArborX::version() << std::endl;
+  std::cout << "ArborX hash       : " << ArborX::gitCommitHash() << std::endl;
+  std::cout << "Kokkos version    : " << KokkosExt::version() << std::endl;
+
+  namespace bpo = boost::program_options;
+
+  int n;
+  bool disallow_loops;
+
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "disallow-loops", bpo::bool_switch(&disallow_loops)->default_value(true), "disallow loops in the graph")
```

Comment:
```suggestion
      ( "allow-loops", bpo::bool_switch(&allow_loops)->default_value(false), "allow loops in the graph")
```
I find it easier to think that `true` means we can have loop instead of `true` means we cannot have loop.